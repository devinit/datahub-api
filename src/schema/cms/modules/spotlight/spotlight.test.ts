import * as prettyFormat from 'pretty-format';
import {getDistrictBySlugAsync, getDistrictEntities} from '.';

describe('spotlight reference module test', () => {

    it('should return uganda district entities', async () => {
        const districts = await getDistrictEntities('uganda');
        expect(prettyFormat(districts)).toMatchSnapshot();
    }, 10000);
    it('should return details about a uganda district', async () => {
        const districts = await getDistrictBySlugAsync('uganda', 'wakiso');
        expect(prettyFormat(districts)).toMatchSnapshot();
    }, 10000);
});
