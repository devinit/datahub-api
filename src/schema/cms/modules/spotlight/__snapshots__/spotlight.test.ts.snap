// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`spotlight reference module test should return details about a uganda district 1`] = `
"Object {
  \\"id\\": \\"d113\\",
  \\"name\\": \\"Wakiso\\",
}"
`;

exports[`spotlight reference module test should return uganda district entities 1`] = `
"Object {
  \\"kenya\\": Array [
    Object {
      \\"id\\": \\"d18943\\",
      \\"name\\": \\"Kiambu\\",
    },
    Object {
      \\"id\\": \\"d18944\\",
      \\"name\\": \\"Kirinyaga\\",
    },
    Object {
      \\"id\\": \\"d18946\\",
      \\"name\\": \\"Murang'a\\",
    },
    Object {
      \\"id\\": \\"d18947\\",
      \\"name\\": \\"Nyandarua\\",
    },
    Object {
      \\"id\\": \\"d18948\\",
      \\"name\\": \\"Nyeri\\",
    },
    Object {
      \\"id\\": \\"d18949\\",
      \\"name\\": \\"Kilifi\\",
    },
    Object {
      \\"id\\": \\"d18950\\",
      \\"name\\": \\"Kwale\\",
    },
    Object {
      \\"id\\": \\"d18951\\",
      \\"name\\": \\"Lamu\\",
    },
    Object {
      \\"id\\": \\"d18952\\",
      \\"name\\": \\"Mombasa\\",
    },
    Object {
      \\"id\\": \\"d18953\\",
      \\"name\\": \\"Taita Taveta\\",
    },
    Object {
      \\"id\\": \\"d18954\\",
      \\"name\\": \\"Tana River\\",
    },
    Object {
      \\"id\\": \\"d18955\\",
      \\"name\\": \\"Embu\\",
    },
    Object {
      \\"id\\": \\"d18956\\",
      \\"name\\": \\"Isiolo\\",
    },
    Object {
      \\"id\\": \\"d18957\\",
      \\"name\\": \\"Kitui\\",
    },
    Object {
      \\"id\\": \\"d18958\\",
      \\"name\\": \\"Machakos\\",
    },
    Object {
      \\"id\\": \\"d18959\\",
      \\"name\\": \\"Makueni\\",
    },
    Object {
      \\"id\\": \\"d18960\\",
      \\"name\\": \\"Marsabit\\",
    },
    Object {
      \\"id\\": \\"d18961\\",
      \\"name\\": \\"Meru\\",
    },
    Object {
      \\"id\\": \\"d18962\\",
      \\"name\\": \\"Nithi\\",
    },
    Object {
      \\"id\\": \\"d18964\\",
      \\"name\\": \\"Nairobi\\",
    },
    Object {
      \\"id\\": \\"d18965\\",
      \\"name\\": \\"Garissa\\",
    },
    Object {
      \\"id\\": \\"d18966\\",
      \\"name\\": \\"Mandera\\",
    },
    Object {
      \\"id\\": \\"d18967\\",
      \\"name\\": \\"Wajir\\",
    },
    Object {
      \\"id\\": \\"d18968\\",
      \\"name\\": \\"Homa Bay\\",
    },
    Object {
      \\"id\\": \\"d18969\\",
      \\"name\\": \\"Kisii\\",
    },
    Object {
      \\"id\\": \\"d18970\\",
      \\"name\\": \\"Kisumu\\",
    },
    Object {
      \\"id\\": \\"d18971\\",
      \\"name\\": \\"Migori\\",
    },
    Object {
      \\"id\\": \\"d18972\\",
      \\"name\\": \\"Nyamira\\",
    },
    Object {
      \\"id\\": \\"d18973\\",
      \\"name\\": \\"Siaya\\",
    },
    Object {
      \\"id\\": \\"d18974\\",
      \\"name\\": \\"Baringo\\",
    },
    Object {
      \\"id\\": \\"d18975\\",
      \\"name\\": \\"Bomet\\",
    },
    Object {
      \\"id\\": \\"d18976\\",
      \\"name\\": \\"Elgeyo-Marakwet\\",
    },
    Object {
      \\"id\\": \\"d18977\\",
      \\"name\\": \\"Kericho\\",
    },
    Object {
      \\"id\\": \\"d18978\\",
      \\"name\\": \\"Laikipia\\",
    },
    Object {
      \\"id\\": \\"d18979\\",
      \\"name\\": \\"Nakuru\\",
    },
    Object {
      \\"id\\": \\"d18980\\",
      \\"name\\": \\"Nandi\\",
    },
    Object {
      \\"id\\": \\"d18981\\",
      \\"name\\": \\"Narok\\",
    },
    Object {
      \\"id\\": \\"d18982\\",
      \\"name\\": \\"Samburu\\",
    },
    Object {
      \\"id\\": \\"d18983\\",
      \\"name\\": \\"Trans-Nzoia\\",
    },
    Object {
      \\"id\\": \\"d18984\\",
      \\"name\\": \\"Turkana\\",
    },
    Object {
      \\"id\\": \\"d18985\\",
      \\"name\\": \\"Uasin Gishu\\",
    },
    Object {
      \\"id\\": \\"d18986\\",
      \\"name\\": \\"West Pokot\\",
    },
    Object {
      \\"id\\": \\"d18987\\",
      \\"name\\": \\"Bungoma\\",
    },
    Object {
      \\"id\\": \\"d18988\\",
      \\"name\\": \\"Busia\\",
    },
    Object {
      \\"id\\": \\"d18989\\",
      \\"name\\": \\"Kakamega\\",
    },
    Object {
      \\"id\\": \\"d18990\\",
      \\"name\\": \\"Vihiga\\",
    },
    Object {
      \\"id\\": \\"d18991\\",
      \\"name\\": \\"Kajiado\\",
    },
  ],
  \\"uganda\\": Array [
    Object {
      \\"id\\": \\"d101\\",
      \\"name\\": \\"Kalangala\\",
    },
    Object {
      \\"id\\": \\"d102\\",
      \\"name\\": \\"Kampala\\",
    },
    Object {
      \\"id\\": \\"d103\\",
      \\"name\\": \\"Kiboga\\",
    },
    Object {
      \\"id\\": \\"d104\\",
      \\"name\\": \\"Luwero\\",
    },
    Object {
      \\"id\\": \\"d105\\",
      \\"name\\": \\"Masaka\\",
    },
    Object {
      \\"id\\": \\"d106\\",
      \\"name\\": \\"Mpigi\\",
    },
    Object {
      \\"id\\": \\"d107\\",
      \\"name\\": \\"Mubende\\",
    },
    Object {
      \\"id\\": \\"d108\\",
      \\"name\\": \\"Mukono\\",
    },
    Object {
      \\"id\\": \\"d109\\",
      \\"name\\": \\"Nakasongola\\",
    },
    Object {
      \\"id\\": \\"d110\\",
      \\"name\\": \\"Rakai\\",
    },
    Object {
      \\"id\\": \\"d111\\",
      \\"name\\": \\"Ssembabule\\",
    },
    Object {
      \\"id\\": \\"d112\\",
      \\"name\\": \\"Kayunga\\",
    },
    Object {
      \\"id\\": \\"d113\\",
      \\"name\\": \\"Wakiso\\",
    },
    Object {
      \\"id\\": \\"d114\\",
      \\"name\\": \\"Lyantonde\\",
    },
    Object {
      \\"id\\": \\"d115\\",
      \\"name\\": \\"Mityana\\",
    },
    Object {
      \\"id\\": \\"d116\\",
      \\"name\\": \\"Nakaseke\\",
    },
    Object {
      \\"id\\": \\"d117\\",
      \\"name\\": \\"Buikwe\\",
    },
    Object {
      \\"id\\": \\"d118\\",
      \\"name\\": \\"Bukomansimbi\\",
    },
    Object {
      \\"id\\": \\"d119\\",
      \\"name\\": \\"Butambala\\",
    },
    Object {
      \\"id\\": \\"d120\\",
      \\"name\\": \\"Buvuma\\",
    },
    Object {
      \\"id\\": \\"d121\\",
      \\"name\\": \\"Gomba\\",
    },
    Object {
      \\"id\\": \\"d122\\",
      \\"name\\": \\"Kalungu\\",
    },
    Object {
      \\"id\\": \\"d123\\",
      \\"name\\": \\"Kyankwanzi\\",
    },
    Object {
      \\"id\\": \\"d124\\",
      \\"name\\": \\"Lwengo\\",
    },
    Object {
      \\"id\\": \\"d201\\",
      \\"name\\": \\"Bugiri\\",
    },
    Object {
      \\"id\\": \\"d202\\",
      \\"name\\": \\"Busia\\",
    },
    Object {
      \\"id\\": \\"d203\\",
      \\"name\\": \\"Iganga\\",
    },
    Object {
      \\"id\\": \\"d204\\",
      \\"name\\": \\"Jinja\\",
    },
    Object {
      \\"id\\": \\"d205\\",
      \\"name\\": \\"Kamuli\\",
    },
    Object {
      \\"id\\": \\"d206\\",
      \\"name\\": \\"Kapchorwa\\",
    },
    Object {
      \\"id\\": \\"d207\\",
      \\"name\\": \\"Katakwi\\",
    },
    Object {
      \\"id\\": \\"d208\\",
      \\"name\\": \\"Kumi\\",
    },
    Object {
      \\"id\\": \\"d209\\",
      \\"name\\": \\"Mbale\\",
    },
    Object {
      \\"id\\": \\"d210\\",
      \\"name\\": \\"Pallisa\\",
    },
    Object {
      \\"id\\": \\"d211\\",
      \\"name\\": \\"Soroti\\",
    },
    Object {
      \\"id\\": \\"d212\\",
      \\"name\\": \\"Tororo\\",
    },
    Object {
      \\"id\\": \\"d213\\",
      \\"name\\": \\"Kaberamaido\\",
    },
    Object {
      \\"id\\": \\"d214\\",
      \\"name\\": \\"Mayuge\\",
    },
    Object {
      \\"id\\": \\"d215\\",
      \\"name\\": \\"Sironko\\",
    },
    Object {
      \\"id\\": \\"d216\\",
      \\"name\\": \\"Amuria\\",
    },
    Object {
      \\"id\\": \\"d217\\",
      \\"name\\": \\"Budaka\\",
    },
    Object {
      \\"id\\": \\"d218\\",
      \\"name\\": \\"Bududa\\",
    },
    Object {
      \\"id\\": \\"d219\\",
      \\"name\\": \\"Bukedea\\",
    },
    Object {
      \\"id\\": \\"d220\\",
      \\"name\\": \\"Bukwo\\",
    },
    Object {
      \\"id\\": \\"d221\\",
      \\"name\\": \\"Butaleja\\",
    },
    Object {
      \\"id\\": \\"d222\\",
      \\"name\\": \\"Kaliro\\",
    },
    Object {
      \\"id\\": \\"d223\\",
      \\"name\\": \\"Manafwa\\",
    },
    Object {
      \\"id\\": \\"d224\\",
      \\"name\\": \\"Namutumba\\",
    },
    Object {
      \\"id\\": \\"d225\\",
      \\"name\\": \\"Bulambuli\\",
    },
    Object {
      \\"id\\": \\"d226\\",
      \\"name\\": \\"Buyende\\",
    },
    Object {
      \\"id\\": \\"d227\\",
      \\"name\\": \\"Kibuku\\",
    },
    Object {
      \\"id\\": \\"d228\\",
      \\"name\\": \\"Kween\\",
    },
    Object {
      \\"id\\": \\"d229\\",
      \\"name\\": \\"Luuka\\",
    },
    Object {
      \\"id\\": \\"d230\\",
      \\"name\\": \\"Namayingo\\",
    },
    Object {
      \\"id\\": \\"d231\\",
      \\"name\\": \\"Ngora\\",
    },
    Object {
      \\"id\\": \\"d232\\",
      \\"name\\": \\"Serere\\",
    },
    Object {
      \\"id\\": \\"d301\\",
      \\"name\\": \\"Adjumani\\",
    },
    Object {
      \\"id\\": \\"d302\\",
      \\"name\\": \\"Apac\\",
    },
    Object {
      \\"id\\": \\"d303\\",
      \\"name\\": \\"Arua\\",
    },
    Object {
      \\"id\\": \\"d304\\",
      \\"name\\": \\"Gulu\\",
    },
    Object {
      \\"id\\": \\"d305\\",
      \\"name\\": \\"Kitgum\\",
    },
    Object {
      \\"id\\": \\"d306\\",
      \\"name\\": \\"Kotido\\",
    },
    Object {
      \\"id\\": \\"d307\\",
      \\"name\\": \\"Lira\\",
    },
    Object {
      \\"id\\": \\"d308\\",
      \\"name\\": \\"Moroto\\",
    },
    Object {
      \\"id\\": \\"d309\\",
      \\"name\\": \\"Moyo\\",
    },
    Object {
      \\"id\\": \\"d310\\",
      \\"name\\": \\"Nebbi\\",
    },
    Object {
      \\"id\\": \\"d311\\",
      \\"name\\": \\"Nakapiripirit\\",
    },
    Object {
      \\"id\\": \\"d312\\",
      \\"name\\": \\"Pader\\",
    },
    Object {
      \\"id\\": \\"d313\\",
      \\"name\\": \\"Yumbe\\",
    },
    Object {
      \\"id\\": \\"d314\\",
      \\"name\\": \\"Abim\\",
    },
    Object {
      \\"id\\": \\"d315\\",
      \\"name\\": \\"Amolatar\\",
    },
    Object {
      \\"id\\": \\"d316\\",
      \\"name\\": \\"Amuru\\",
    },
    Object {
      \\"id\\": \\"d317\\",
      \\"name\\": \\"Dokolo\\",
    },
    Object {
      \\"id\\": \\"d318\\",
      \\"name\\": \\"Kaabong\\",
    },
    Object {
      \\"id\\": \\"d319\\",
      \\"name\\": \\"Koboko\\",
    },
    Object {
      \\"id\\": \\"d320\\",
      \\"name\\": \\"Maracha\\",
    },
    Object {
      \\"id\\": \\"d321\\",
      \\"name\\": \\"Oyam\\",
    },
    Object {
      \\"id\\": \\"d322\\",
      \\"name\\": \\"Agago\\",
    },
    Object {
      \\"id\\": \\"d323\\",
      \\"name\\": \\"Alebtong\\",
    },
    Object {
      \\"id\\": \\"d324\\",
      \\"name\\": \\"Amudat\\",
    },
    Object {
      \\"id\\": \\"d325\\",
      \\"name\\": \\"Kole\\",
    },
    Object {
      \\"id\\": \\"d326\\",
      \\"name\\": \\"Lamwo\\",
    },
    Object {
      \\"id\\": \\"d327\\",
      \\"name\\": \\"Napak\\",
    },
    Object {
      \\"id\\": \\"d328\\",
      \\"name\\": \\"Nwoya\\",
    },
    Object {
      \\"id\\": \\"d329\\",
      \\"name\\": \\"Otuke\\",
    },
    Object {
      \\"id\\": \\"d330\\",
      \\"name\\": \\"Zombo\\",
    },
    Object {
      \\"id\\": \\"d401\\",
      \\"name\\": \\"Bundibugyo\\",
    },
    Object {
      \\"id\\": \\"d402\\",
      \\"name\\": \\"Bushenyi\\",
    },
    Object {
      \\"id\\": \\"d403\\",
      \\"name\\": \\"Hoima\\",
    },
    Object {
      \\"id\\": \\"d404\\",
      \\"name\\": \\"Kabale\\",
    },
    Object {
      \\"id\\": \\"d405\\",
      \\"name\\": \\"Kabarole\\",
    },
    Object {
      \\"id\\": \\"d406\\",
      \\"name\\": \\"Kasese\\",
    },
    Object {
      \\"id\\": \\"d407\\",
      \\"name\\": \\"Kibaale\\",
    },
    Object {
      \\"id\\": \\"d408\\",
      \\"name\\": \\"Kisoro\\",
    },
    Object {
      \\"id\\": \\"d409\\",
      \\"name\\": \\"Masindi\\",
    },
    Object {
      \\"id\\": \\"d410\\",
      \\"name\\": \\"Mbarara\\",
    },
    Object {
      \\"id\\": \\"d411\\",
      \\"name\\": \\"Ntungamo\\",
    },
    Object {
      \\"id\\": \\"d412\\",
      \\"name\\": \\"Rukungiri\\",
    },
    Object {
      \\"id\\": \\"d413\\",
      \\"name\\": \\"Kamwenge\\",
    },
    Object {
      \\"id\\": \\"d414\\",
      \\"name\\": \\"Kanungu\\",
    },
    Object {
      \\"id\\": \\"d415\\",
      \\"name\\": \\"Kyenjojo\\",
    },
    Object {
      \\"id\\": \\"d416\\",
      \\"name\\": \\"Buliisa\\",
    },
    Object {
      \\"id\\": \\"d417\\",
      \\"name\\": \\"Ibanda\\",
    },
    Object {
      \\"id\\": \\"d418\\",
      \\"name\\": \\"Isingiro\\",
    },
    Object {
      \\"id\\": \\"d419\\",
      \\"name\\": \\"Kiruhura\\",
    },
    Object {
      \\"id\\": \\"d420\\",
      \\"name\\": \\"Buhweju\\",
    },
    Object {
      \\"id\\": \\"d421\\",
      \\"name\\": \\"Kiryandongo\\",
    },
    Object {
      \\"id\\": \\"d422\\",
      \\"name\\": \\"Kyegegwa\\",
    },
    Object {
      \\"id\\": \\"d423\\",
      \\"name\\": \\"Mitooma\\",
    },
    Object {
      \\"id\\": \\"d424\\",
      \\"name\\": \\"Ntoroko\\",
    },
    Object {
      \\"id\\": \\"d425\\",
      \\"name\\": \\"Rubirizi\\",
    },
    Object {
      \\"id\\": \\"d426\\",
      \\"name\\": \\"Sheema\\",
    },
  ],
}"
`;
